src/standalone/Vegetation/soil_drivers.jl

```diff
@@ -0,0 +1,35 @@
+using ..ClimaLSM.Soil
+export AbstractSoilDriver, PrescribedSoil, PrognosticSoil
+
+"""
+     An abstract type of soil drivers of the canopy model.
+"""
+abstract type AbstractSoilDriver{FT <: AbstractFloat} end
+
+"""
+     PrescribedSoil{} <: AbstractSoilDriver
+
+A container for holding prescribed soil parameters needed by the canopy model
+when running the canopy in standalone mode, including the soil pressure and 
+albedo.
+$(DocStringExtensions.FIELDS)
+"""
+struct PrescribedSoil{FT} <: AbstractSoilDriver{FT}
+    "The depth of the root tips, in meters"
+    root_depths::Vector{FT}
+    "Prescribed soil potential (m) as a function of time"
+    ψ_soil::Function
+    "Soil albedo"
+    soil_α::FT
+end
+
+"""
+   PrognosticSoil{FT} <: AbstractSoilDriver{FT}
+
+Concrete type of AbstractSoilDriver used for dispatch in cases where both 
+a canopy model and soil model are run. Contains the soil albedo to be shared 
+between the canopy and soil models
+"""
+struct PrognosticSoil{FT} <: AbstractSoilDriver{FT}
```

Comment:
What do you think about defining an outer constructor for this which takes this as a keyword argument? it might be a little clearer in usage.

I wonder if we could also add to the docstring something like this:

"When running the SoilCanopyModel, the soil model specifies the albedo. In this case, the constructor for the model reads the albedo from the soil model and the user only specifies it once, for the soil model. When running the SoilPlantHydrologyModel - which is intended for internal usage/testing primarily - the user must specify the albedo because the soil model in that case does not have an albedo."

I dont know if that would make it more confusing :P but im just thinking about how to keep all of this clear.